specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/DynamicSummaries.json

```diff
@@ -0,0 +1,367 @@
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/SummaryId"
+          },
+          {
+            "$ref": "#/parameters/Summary"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Summary"
+            }
+          },
+          "201": {
+            "description": "Created. The response includes the Provisioning State and the Azure-AsyncOperation header. To get the progress of the operation, call GET operation on the URL in Azure-AsyncOperation header field.",
+            "schema": {
+              "$ref": "#/definitions/Summary"
+            },
+            "headers": {
+              "Azure-AsyncOperation": {
+                "description": "Contains the status URL on which clients are expected to poll the status of the operation.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SummaryList": {
+      "description": "List all the summaries.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of summaries.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of summaries.",
+          "items": {
+            "$ref": "#/definitions/Summary"
+          },
+          "type": "array"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "Summary": {
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
+        }
+      ],
+      "description": "Represents a Summary in Azure Security Insights.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SummaryProperties",
+          "description": "Summary properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object"
+    },
+    "SummaryProperties": {
+      "description": "Describes summary properties",
+      "properties": {
+        "summaryId": {
+          "description": "The id (a Guid) of the summary",
+          "type": "string"
+        },
+        "summaryName": {
```

Comment:
>summaryName [](http://example.com/codeflow?start=9&length=11)

top level envelope already has a name

Reply:
not sure if I understood the issue here. the DynamicSummary resource has a property "summaryName" which user defines that (and it is editable) to name the summary. the summary has summaryId which is in GUID format and is uniqe and not editable

Reply:
+1. This an ARM resource, which means the top level properties already include id and name (defined in the ResourceWithEtag $ref above). You shouldn't duplicate those properties within the property bag. If summaryId is different from the top-level "id" property and/or summaryName is different from the top-level "name" property, then these should get different names to make it clear what's different about them.